src/javascript/app/common/form_validation.js

```diff
@@ -363,15 +364,15 @@ const Validation = (() => {
         if (!all_is_ok) {
             showError(field, message);
         } else {
-            clearError(field);
+            clearError(field, message);
         }
 
         return all_is_ok;
     };
 
-    const clearError = (field) => {
+    const clearError = (field, localized_message) => {
         if (field.$error && field.$error.length) {
-            field.$error.setVisibility(0);
+            field.$error.html(localized_message).setVisibility(0);
```

Comment:
I'm a bit confused... why do we need to set the error message before the message is cleared?

Reply:
the showError function uses the setVisibility functions on the .html() contents that is equal to localized_message. In order for the error message to be properly cleared, the clearError function needs to be the same.

If the setVisibility function is set on the $error element itself, the error message will stay invisible if it gets triggered, because the showError function is set on the contents of this element '.html(localized_message)', not the element itself like clearError.